src/components/pages/address/address-transactions.tsx

```diff
@@ -1,13 +1,66 @@
 import { Address } from "viem";
+import { prisma } from "@/lib/prisma";
 import { Card, CardHeader, CardContent } from "@/components/ui/card";
+import { fromPrismaTransactionWithReceipt, TransactionWithReceipt } from "@/lib/types";
+import { loadFunctions } from "@/lib/signatures";
+import TxsTable from "@/components/lib/txs-table";
 
-const AddressTransactions = ({ address }: { address: Address }) => {
+const TRANSACTION_LIMIT = 25;
+interface TransactionQueryResult {
+  transactions: TransactionWithReceipt[];
+  totalCount: number;
+}
+
+const fetchTransactionsData = async (address: Address): Promise<TransactionQueryResult> => {
+  const associatedTransactions = await prisma.transaction.findMany({
+    where: {
+      OR: [
+        { from: { contains: address } },
+        { to: { contains: address } },
+      ],
+    },
+    orderBy: { timestamp: 'desc' },
+    include: { receipt: true },
+  });
+  
+//Filter null reciept
+  const transactionsWithReceipts = associatedTransactions.filter(tx => tx.receipt !== null);
+  const validTransactions = await Promise.all(
+    transactionsWithReceipts.map(async (tx) => {
+      try {
+        const signature = await loadFunctions(tx.input.slice(0, 10));
+        return fromPrismaTransactionWithReceipt(tx, tx.receipt!, signature);
+      } catch (error) {
+        console.error(`Error processing transaction ${tx.hash}:`, error);
+        return null;
+      }
+    })
+  );
+
+  const filteredValidTransactions = validTransactions.filter(tx => tx !== null) as TransactionWithReceipt[];
+  return {
+    transactions: filteredValidTransactions.slice(0, TRANSACTION_LIMIT),
+    totalCount: filteredValidTransactions.length,
+  };
+};
+
+const AddressTransactions: React.FC<{ address: Address }> = async ({ address }) => {
+  const { transactions, totalCount } = await fetchTransactionsData(address);
   return (
     <Card>
-      <CardHeader>Latest 25 from a total of N transactions</CardHeader>
-      <CardContent>TRANSACTIONS</CardContent>
+      <CardHeader>
+        Latest {transactions.length >= TRANSACTION_LIMIT ? TRANSACTION_LIMIT : transactions.length} from a total of {totalCount} valid transactions
```

Comment:
You can safely use transactions.length here after making the rest of the requested changes.